### PR TITLE
Fix typedefs with container types

### LINKED ISSUE #111: Typedef set causes an error
The following thrift snippet:

```
  typedef set<i32> SetOfIntegers

  struct Thangs {
    1: optional SetOfIntegers magic_numbers
  }
```

Causes the following error on `mix thrift.generate`:

```
** (exit) an exception was raised:
    ** (CaseClauseError) no case clause matching: {:set, :i32}
        (thrift) lib/thrift/parser/resolver.ex:37: anonymous fn/2 in Thrift.Parser.Resolver.update/3
        (elixir) lib/map.ex:114: Map.do_new_transform/3
        (thrift) lib/thrift/parser/resolver.ex:36: Thrift.Parser.Resolver.update/3
        (elixir) lib/agent/server.ex:23: Agent.Server.handle_call/3
        (stdlib) gen_server.erl:615: :gen_server.try_handle_call/4
        (stdlib) gen_server.erl:647: :gen_server.handle_msg/5
        (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
```

### PR DESCRIPTION
**Pull request by scohen**
Fixes #111

### CODE REVIEW

File: lib/thrift/parser/resolver.ex [hunk 1]
```diff
@@ -36,6 +36,8 @@ defmodule Thrift.Parser.Resolver do
     new_mappings = local_mappings
     |> Map.new(fn {name, val} ->
       case val do
+        val when is_tuple(val) ->
+          {:"#{include_name}.#{name}", val}
         val when is_atom(val) ->
```
Comment (jparise) [inline lib/thrift/parser/resolver.ex]: Could we also do `val when is_atom(val) or is_tuple(val)` and keep this one case?

File: test/generator/models_test.exs [hunk 2]
```diff
@@ -149,4 +149,16 @@ defmodule Thrift.Generator.ModelsTest do
...
+  @thrift_file name: "complex_typedefs.thrift", contents: """
+  typedef set<i32> IntSet
+  struct Thangs {
+    1: optional IntSet numbers
+  }
+  """
+
+  thrift_test "it should be able to have typedefs" do
+    thang = %Thangs{numbers: MapSet.new([1, 2, 3])}
+    assert thang.numbers == MapSet.new([1, 2, 3])
```
Comment (dantswain) [inline test/generator/models_test.exs]: Would it be sufficient to `assert %Thangs{numbers: MapSet.new([1, 2, 3])}`?  The assertion here is really just asserting that struct access works.

Comment (scohen) [inline test/generator/models_test.exs]: this is asserting that the field exists. Yes, it's pedantic.

Comment (dantswain) [inline test/generator/models_test.exs]: If the field didn't exist then `%Thangs{numbers: MapSet.new([1, 2, 3])}` would fail, too, wouldn't it?

**Pull request merged**